core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1381,7 +1391,8 @@ private case class OptionAssigner(
     clusterManager: Int,
     deployMode: Int,
     clOption: String = null,
-    confKey: String = null)
+    confKey: String = null,
+    mergeFn: Option[(String, String) => String] = None)
```

Comment:
This needs to receive a function, since how to merge two values depends on the semantic of the conf value.